lib/secure-boot-lib-common.robot

```diff
@@ -0,0 +1,122 @@
+
+Library             ../lib/secure-boot-lib-common.py
+Resource            ../keywords.robot
+
+
+*** Keywords ***
+Check If Certificate Images For Tests Exists
+    [Documentation]    This keyword generates all the necessary images
+    ...    for secure boot tests.
+    ${images_list}=    Create List    BAD_FORMAT    BAD_KEYS    ECDSA256    ECDSA384
+    ...    ECDSA521    EXPIRED    GOOD_KEYS    INTERMEDIATE    NOT_SIGNED
+    ...    RSA2048    RSA3072    RSA4096
+    FOR    ${image}    IN    @{images_list}
+        ${image_path}=    Set Variable    ${CURDIR}/../scripts/secure-boot/images/${image}.img
+        OperatingSystem.File Should Exist
+        ...    ${image_path}
+        ...    Image ${image}.img does not exist! Please run ./scripts/secure-boot/generate-images/sb-img-wrapper.sh script.
+    END
+
+Check Secure Boot In Linux
+    [Documentation]    Keyword checks Secure Boot state in Linux.
+    ...    Returns True when Secure Boot is enabled
+    ...    and False when disabled.
+    # The string in dmesg may be in two forms:
+    # secureboot: Secure boot disabled
+    # or just:
+    # Secure boot disabled
+    ${out}=    Execute Command In Terminal    dmesg | grep "Secure boot"
+    Should Contain Any    ${out}    disabled    enabled
+    ${sb_status}=    Run Keyword And Return Status
+    ...    Should Contain    ${out}    enabled
+    RETURN    ${sb_status}
+
+Check Secure Boot In Windows
+    [Documentation]    Keyword checks Secure Boot state in Windows.
+    ...    Returns True when Secure Boot is enabled
+    ...    and False when disabled.
+    ${out}=    Execute Command In Terminal    Confirm-SecureBootUEFI
+    Should Contain Any    ${out}    True    False
+    ${sb_status}=    Run Keyword And Return Status
+    ...    Should Contain    ${out}    True
+    RETURN    ${sb_status}
+
+Autoenroll Secure Boot Certificates
+    [Documentation]    Enrolls Secure boot certificates automatically using tool
+    ...    for automatic provisioning
+    # 1. Erase Secure Boot Keys
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    ${key_menu}=    Enter Key Management And Return Construction    ${sb_menu}
+    Erase All Secure Boot Keys    ${key_menu}
+    Save Changes And Reset    3    5
+
+    # 2. Boot to Automatic provisioning tool
+    Boot Dasharo Tools Suite    USB    ${TRUE}
+
+    # 3. Enable SB after provisioning
+    ${sb_menu}=    Enter Secure Boot Menu And Return Construction
+    Enable Secure Boot    ${sb_menu}
+    Save Changes And Reset
+
+Remove Old Secure Boot Keys In OS
+    [Documentation]    Removes all files and directories in
+    ...    `/usr/share/secureboot` which is used by sbctl in the operating
+    ...    system
+    Execute Linux Command Without Output    rm -rf /usr/share/secureboot
+
+Generate Secure Boot Keys In OS
+    [Documentation]    Generates new Secure Boot keys using sbctl.
+    ...    Returns true if succeeded.
+    ${out}=    Execute Linux Command    sbctl create-keys
```

Comment:
Use Terminal kwds. In the whole MR.

Reply:
Added but this was not tested yet https://github.com/Dasharo/open-source-firmware-validation/pull/187/commits/1b85aa7b649fa94a881a79f22f2f646bd04d18d4

I think that whole MR was checked and this KWD is not used in it.